### PR TITLE
Tlon desktop: move Talk functionality into main app + sidebar updates

### PR DESCRIPTION
**Pull request by latter-bolden**
Adds the ability to access the core Talk sidebar from within the main app on desktop. Overhauls the way the Sidebar looks, adds Search for group/message lists, adds flow for desktop DM creation, etc.
Tested locally against self & hosted ships.

Fixes LAND-1513
Fixes LAND-1522

### CODE REVIEW

File: ui/src/components/Sidebar/useSearchFilter.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  //         ? b.score + 1
+  //         : b.score;
+
+  //       return right - left;
+  //     })
+  // ), [query, searchSpace]);
+
+  // const results = useMemo(() => fuzzScores.map(result => records[result.index]), [fuzzScores, records]);
+
+  // return results;
```
Comment (arthyn) [inline ui/src/components/Sidebar/useSearchFilter.ts:78]:
```suggestion
```

File: ui/src/components/Sidebar/useSearchFilter.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      [...groupsToRecords(groups), ...gangsToRecords(gangs)],
+      comparator
+    );
+  }, [groups, gangs]);
+
+  if (!query) {
+    return [];
+  }
+
+  return searchSpace.filter(
```
Comment (arthyn) [inline ui/src/components/Sidebar/useSearchFilter.ts:52]: we could use fuzzy here maybe, but this is fine for now

File: ui/src/components/Sidebar/MessagesSidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+              return false;
+            }}
+          >
+            <ArrowNWIcon className="text-gray-400" />
+          </a>
+        </div>
+      </SidebarItem>
+    </ActionMenu>
+  );
+}
```
Comment (arthyn) [inline ui/src/components/Sidebar/MessagesSidebar.tsx:127]: unused

**Review by arthyn — APPROVED** [review]
lgtm, a few suggestions/comments

File: ui/src/components/Sidebar/useSearchFilter.ts (hunk 2 above)
Comment (latter-bolden) [reply ui/src/components/Sidebar/useSearchFilter.ts:52]: ya fell back to filter cause fuzz was slowing it down a bit

Comment (arthyn) [reply ui/src/components/Sidebar/useSearchFilter.ts:52]: ahh gotcha cool

**Comment by mrozanski** [comment]
Found while testing this branch:

Desktop, Groups, chat, Clicking on a sigil of a message's author
Expected:
stay in the same browser tab/window, open DM UI in the Messages app nav section
For example: https://localhost:3000/apps/groups/dm/~nocsyx-lassul
Observed:
New browser  tab pointing to https://localhost:3000/apps/talk/dm/~nocsyx-lassul

Note: Desktop browser in small (mobile) size does the same.
Haven't tested from the mobile app.

**Comment by mrozanski** [comment]
I’ve noticed that when I click on an item in Activity that opens a group chat, if then I click the back arrow at the top of the side nav, I’m taken to the Groups home with no contents in the main pane. I would expect to go back to Activity instead.

On mobile and if the activity belongs to a thread this is more noticeable because clicking on `<` sends me to the main thread of that channel, then to the group home, and then to groups.
The behavior is probably designed like this and it would make sense that I can navigate back that way in other circumstances. But when coming from the activity list, I personally prefer to go back to that list in one step, as I’d most likely be catching up with activity in general.

**Comment by mrozanski** [comment]
I assume it is expected (found the same in Figma) but noting that the Groups icon is now a house instead of the one currently being used.
group-icon

**Comment by mrozanski** [comment]
Messages > Filter icon next to Direct Messages (drop down)
"Group Talk Channels" doesn't sound right.

grouptalk

Maybe "Group channels"?

**Comment by mrozanski** [comment]
Member links in Chat info point to talk URL

Opening a multi-DM, clicking on the three dots at the top > Chat info

multi-dm-info

**Review by mrozanski — APPROVED** [review]
Comments verified.
Approving.

**Pull request merged**